tests/cunit/test_pioc.c

```diff
@@ -665,12 +739,17 @@ int test_names(int iosysid, int num_flavors, int *flavor, int my_rank,
             ERR(ret);
 
         /* Define a global attribute. */
-        int att_val = 42;
+        int att_val = ATT_VAL;
+        float float_att_val = ATT_VAL;
         if ((ret = PIOc_put_att_int(ncid, NC_GLOBAL, ATT_NAME, PIO_INT, 1, &att_val)))
             ERR(ret);
+        if ((ret = PIOc_put_att_float(ncid, NC_GLOBAL, ATT_NAME2, PIO_FLOAT, 1, &float_att_val)))
+            ERR(ret);
+        /* if ((ret = PIOc_put_att_int(ncid, NC_GLOBAL, ATT_NAME2, PIO_FLOAT, 1, &att_val))) */
```

Comment:
That's how I discovered that conversions did not work. ;-)

I have removed that line in my current branch. It's not needed. I am adding systematic tests for attribute type conversion.